addons/api/addon/serializers/credential.js

```diff
@@ -13,6 +13,20 @@ export default class CredentialSerializer extends ApplicationSerializer {
    */
   serializeScopeID = false;
 
+  serialize() {
+    const serialized = super.serialize(...arguments);
+
+    if (serialized.type === 'json') {
+      delete serialized.attributes.username;
+
+      serialized.attributes.object = serialized.attributes.json_object;
+      delete serialized.attributes.json_object;
+      let parsedSecret = JSON.parse(serialized.attributes.object);
+      serialized.attributes.object = parsedSecret;
```

Comment:
You could've just parsed this first and stored it in `serialized.attributes.object` rather than saving to `serialized.attributes.object` twice.

e.g. `serialized.attributes.object = JSON.parse(serialized.attributes.json_object)`

Reply:
thansk @ZedLi fixed